### DISCUSSION TITLE
Overwrite global Datatable filter

### DISCUSSION THREAD

**Discussion by sja-cslab** [primefaces/community#1642] [PrimeReact] [answered]
Currently I'm trying to overwrite the global filter on a Datatable the following way

```js
const [filters, setFilters] = useState({
        global: {
            value: null,
            matchMode: "multiwithwhitespace"
        }
    });
[...]

FilterService.register("multiwithwhitespace", (value, filter) => { [...]
```
That seems to work, even if I'm not quite sure if its intended.
However I noticed, that `filterDelay` seems to be ignored.

What am i doing wrong here? Should I even try to overwrite the global filter?

**Comment by sja-cslab** [comment]
@melloware i bet you can answer me ;)

**Reply by melloware** [reply, marked as answer]
Hmm I have not messed with custom filters before but your global filter uses your own `<inputText>` right?  You can just add the `useDebouce` https://primereact.org/hooks/usedebounce/ to add a filter delay?

**Reply by sja-cslab** [reply]
Yea its my own input. Good hint, didn't try.
Thank you, i'll give it a shot.
